overlord/devicestate/firstboot_preseed_test.go

```diff
@@ -191,6 +187,47 @@ func checkPreseedOrder(c *C, tsAll []*state.TaskSet, snaps ...string) {
 			c.Check(waitTasks[0], Equals, prevTask)
 		}
 
+		// make sure that install-hooks wait for the previous snap, and for
+		// mark-preseeded.
+		hookEdgeTask, err := ts.Edge(snapstate.HooksEdge)
+		c.Assert(err, IsNil)
+		c.Assert(hookEdgeTask.Kind(), Equals, "run-hook")
+		var hsup hookstate.HookSetup
+		c.Assert(hookEdgeTask.Get("hook-setup", &hsup), IsNil)
+		c.Check(hsup.Hook, Equals, "install")
+		if hsup.Snap != "core" && hsup.Snap != "core18" && hsup.Snap != "snapd" {
```

Comment:
super minor nitpick, this could be a little more readable as a switch, i.e.

```suggestion
		switch hsup.Snap {
		case "core", "core18", "snapd":
			// do the things
		}
```